script.js

```diff
@@ -16,28 +16,38 @@ const shuffleArray = (array) => {
 // Lets try to keep the global variables to a minimum.
 
 // some constants:
-const ANIMATION_DURATION = 200;
-// pairCount shouldn't be bigger than the number of Object.keys(DATA).length, otherwise only words will be rendered
-let pairCount = 3;
+const ANIMATION_DURATION = 250;
 
-// We are storing the reference to the last clicked divs in the global scope, so we can access them from any function.
-let firstClicked = null;
-let secondClicked = null;
-
-// Get the list of english words from the dictionary and shuffle them (we want each game to be different)
-const shuffledKeys = shuffleArray(Object.keys(DATA));
-// safety check - if desired pairCount >= shuffledKeys.length -> pairCount = shuffledKeys.length - 1
-if (pairCount >= shuffledKeys.length) {
-    pairCount = shuffledKeys.length;
+const totalWordsInSessionCount = 8;
+// pairsToRenderCount shouldn't be bigger than the totalWordsInSessionCount, otherwise only words will be rendered
+let pairsToRenderCount = 3;
+// safety check - if desired pairsToRenderCount >= shuffledKeys.length -> pairsToRenderCount = shuffledKeys.length - 1
+if (pairsToRenderCount > totalWordsInSessionCount) {
+    pairsToRenderCount = totalWordsInSessionCount;
 }
-// Create the pairs of words and kanjis
-const wordPairs = shuffledKeys.map((word) => [word, DATA[word]]);
+
+// We are storing the reference to the last clicked divs in the global scope, so we can access them from any function.
+let kanjiClicked = null;
+let hiraganaClicked = null;
+
 // keep track of the last used pair
 let lastUsedPairIndex = 0;
-// kep track of the number of found pairs
+// keep track of the number of found pairs
 let foundPairs = 0;
+// variables for starting the timer and incrementing the time
+let gameStart;
+let timerInterval;
 // -------------------
 
+const kanjiHiraganaGlossary = [];
+for(let i = 0; i < totalWordsInSessionCount; i++) {
+  const kanji = JM[i].kanji;
+  const hiragana = JM[i]['hiragana/katakana'];
+  const glossary = JM[i].glossary;
+  kanjiHiraganaGlossary.push([kanji, hiragana, glossary]);
+}
```

Comment:
lets create a function for this logic - one should avoid logic in the global scope. Keep in mind that you will have to write tests for this app, and the easiest thing for testing is a plain function. The tests will look somehow like this:
```javascript
import { createKanjiHiraganaGlossary } from './functions.js';

test('should create glossary when input data is correct', () => {
  // Arrange
  const JM = { 
     // ...
  };
  const totalWords = 10;
  // Act
  const result = createKanjiHiraganaGlossary(JM, totalWords);
  // Assert
  expect(result).toMatchObject({ 
     // ...
  });
});
```
whereas if you leave logic in the global scope... good luck with testing :sweat_smile: